src/SSEConnection.js

```diff
@@ -0,0 +1,54 @@
+import AbstractConnection from "./AbstractConnection";
+import ConnectionEvent from "./ConnectionEvent";
+
+/**
+ * Server Sent Event Connections
+ */
+export default class SSEConnection extends AbstractConnection {
+  constructor(url, options = {}) {
+    super(url);
+    this.options = options;
+    this.source = null;
+    this.data = [];
+  }
+
+  open() {
+    const { withCredentials = false } = this.options;
+    this.source = new window.EventSource(this.url, { withCredentials });
+
+    this.source.onopen = () => {
+      this.state = SSEConnection.OPEN;
+      this.emit(
+        ConnectionEvent.OPEN,
+        new ConnectionEvent(this, ConnectionEvent.OPEN)
+      );
+    };
+
+    this.source.onmessage = response => {
+      this.data.push(response);
+      this.emit(
+        ConnectionEvent.DATA,
+        new ConnectionEvent(this, ConnectionEvent.DATA)
+      );
+    };
+
+    this.source.onerror = error => {
+      this.state = SSEConnection.CLOSED;
+      this.source.close();
+      this.error = error;
+      this.emit(
+        ConnectionEvent.ERROR,
+        new ConnectionEvent(this, ConnectionEvent.ERROR)
+      );
+    };
+  }
+
+  close() {
+    this.state = SSEConnection.CLOSED;
+    this.source.close();
+    this.emit(
+      ConnectionEvent.ABORT,
```

Comment:
since there is no way to distinguish between "abort" and "complete", I'd replace `ABORT` with `COMPLETE` here 🤔

Reply:
Sure, they seem pretty interchangeable in this context